src/libs/core/extractor.cpp

```diff
@@ -56,7 +56,7 @@ void Extractor::extract(const QString &sourceFile, const QString &destination, c
 
     QDir destinationDir(destination);
     if (!root.isEmpty()) {
-        destinationDir = destinationDir.filePath(root);
+        destinationDir.setPath(destinationDir.filePath(root));
```

Comment:
Unrelated to the rest of the patch, what's the reason for this?

Reply:
I mentioned it in my first statement. `QDir::operator=(const QDir&)` has gone deprecated too.
I strangely [don't see this reported](https://doc.qt.io/qt-5/qdir.html#operator-eq) on the website though. Here's the compiler warning:
```
../src/libs/core/extractor.cpp: In member function ‘void Zeal::Core::Extractor::extract(const QString&, const QString&, const QString&)’:
../src/libs/core/extractor.cpp:59:54: warning: ‘QDir& QDir::operator=(const QString&)’ is deprecated: Use QDir::setPath() instead [-Wdeprecated-declarations]
   59 |         destinationDir = destinationDir.filePath(root);
      |                                                      ^
In file included from /usr/include/qt/QtCore/QDir:1,
                 from ../src/libs/core/extractor.cpp:25:
/usr/include/qt/QtCore/qdir.h:110:11: note: declared here
  110 |     QDir &operator=(const QString &path);
      |           ^~~~~~~~
```

Reply:
I didn't see the mention of `QDir::operator=(const QString&)` until now, lol. Merging, thanks!